kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/ServiceControlCliArguments.scala

```diff
@@ -137,200 +212,105 @@ class ServiceControlCliArguments(args: Seq[String], env: Map[String, String] = s
   }
 
   private def validateHostAndPort(): Unit = {
-    if (host == null) {
+    if (cliArgs.host == null) {
       fail("Must specify host for service")
     }
-    if (port == null) {
+    if (cliArgs.port == null) {
       fail("Must specify port for service")
     }
 
     try {
-      host = InetAddress.getByName(host).getCanonicalHostName
+      cliArgs = cliArgs.copy(host = InetAddress.getByName(cliArgs.host).getCanonicalHostName)
     } catch {
       case _: Exception =>
-        fail(s"Unknown host: $host")
+        fail(s"Unknown host: ${cliArgs.host}")
     }
 
     try {
-      if (port.toInt <= 0 ) {
+      if (cliArgs.port.toInt <= 0 ) {
         fail(s"Specified port should be a positive number")
       }
     } catch {
       case _: NumberFormatException =>
-        fail(s"Specified port is not a valid integer number: $port")
+        fail(s"Specified port is not a valid integer number: ${cliArgs.port}")
     }
   }
 
   private def validateUser(): Unit = {
-    if (service == ServiceControlObject.ENGINE && user == null) {
-      fail("Must specify user name for engine")
+    if (cliArgs.service == ServiceControlObject.ENGINE && cliArgs.user == null) {
+      fail("Must specify user name for engine, please use -u or --user.")
     }
   }
 
   private[ctl] def printUsageAndExit(exitCode: Int, unknownParam: Any = null): Unit = {
     if (unknownParam != null) {
-      info("Unknown/unsupported param " + unknownParam)
+      info(unknownParam)
     }
     val command =
       s"""
         |Kyuubi Ver $KYUUBI_VERSION.
-        |Usage: kyuubi-ctl <create|get|delete|list>  <server|engine> --zk-quorum ...
+        |Usage: kyuubi-ctl [create|get|delete|list]  [server|engine] --zk-quorum ...
         |--namespace ... --user ... --host ... --port ... --version""".stripMargin
     info(command)
 
     info(
       s"""
          |Command:
```

Comment:
Can scopt help generate the usage message automatically

Reply:
> Can scopt help generate the usage message automatically

Yes, but i think it is not handsome enough and is too long.
In addition,  `service` show in `Usage` help as a option of `action`(`Usage: kyuubi-ctl [create|get|delete|list] [options]`), 
`Usage: kyuubi-ctl [create|get|delete|list]  [server|engine] --zk-quorum` looks like better.
Scopt generate help like:
```
Usage: kyuubi-ctl [create|get|delete|list] [options]

Command: create [server|engine] [options]

Command: create server
    .
    .
    .
  --help
        
Command: create engine

  -zk <value> | --zookeeper <value>
    .
    .
    .
   --help
Command: get [server|engine] [options]

Command: get server

  -zk <value> | --zookeeper <value>
    .
    .
    .
  --help
    .
    .
    .
```

Reply:
I am ok with the scopt output. We gonna accept it if we want to introduce it into Kyuubi. Also, it reduces the burden of codebase maintenance.

WDYT? @pan3793  @turboFei

Reply:
`scopt` output LGTM with some modification.

Here is a demo
```
➜  bin ./clickhouse-tool --help
ClickHouse Backup Tool
Usage: clickhouse-tool [dump|restore] [options]

  -h, --help               prints this usage text
  -v, --verbose
  --dry-run
  --host <host/ip>
  --port <tcp port>
  --username <username>
  --password <password>
  --data-dir <dir>         directory to store data
  --databases <db1,db2...>
                           databases to dump/restore, all if absent
  --tables <tbl1,tbl2...>  tables to dump/restore, all if absent. if present, must specify one database


Command: dump [options]

  --overwrite              overwrite if data-dir exist. default false
  --format <format>        JSONEachRow(default), Parquet, Native

Command: restore [options]

  --on-cluster <cluster>   run DML on cluster, only run on single node if absent
  --skip-create-table      skip run create table sql
  --create-table-if-not-exists
                           run create table sql, do nothing if table exist
  --truncate-table         truncate table before insert records

Alias:
  clickhouse-dump    => clickhouse-tool dump
  clickhouse-restore => clickhouse-tool restore
```

Reply:
Tips: use `note(string)` to insert custom content in help text.

```
note("")
    note("Alias:")
    note("  clickhouse-dump    => clickhouse-tool dump")
    note("  clickhouse-restore => clickhouse-tool restore")
```

Reply:
I am OK with scopt be involved into kyuubi.
The output showed by pan above LGTM.
It is better to optimize the usage output, it is important.

Reply:
Agree, `Usage` is important. Since everyone can accept scopt's output, i'll try to use and make it show better.